packages/library-base/src/commonMain/kotlin/io/realm/internal/RealmObjectInternal.kt

```diff
@@ -91,18 +96,43 @@ interface RealmObjectInternal : RealmObject, RealmStateHolder, io.realm.internal
     override fun emitFrozenUpdate(
         frozenRealm: RealmReference,
         change: NativePointer,
-        channel: SendChannel<RealmObjectInternal>
+        channel: SendChannel<ObjectChange<RealmObjectInternal>>
     ): ChannelResult<Unit>? {
-        val f: RealmObjectInternal? = this.freeze(frozenRealm)
-        return if (f == null) {
+        val frozenObject: RealmObjectInternal? = this.freeze(frozenRealm)
+
+        return if (frozenObject == null) {
+            channel.trySend(DeletedObjectImpl())
             channel.close()
             null
```

Comment:
Shouldn't we return the result of `trySend` here? Not 100% sure what receives of this do with it

Reply:
Yes, we must return the result of the trysend so the notifier can evaluate its result.